### ISSUE TITLE
Server crash: java.lang.Error: Watchdog

### ISSUE THREAD

**Issue by wereii** [ElectroluxV2/git-backup-fabric-mod#1]
Minecraft Version: 1.20.1
Operating System: Linux (amd64) version 5.10.0-23-amd64
Java Version: 17.0.8, Eclipse Adoptium

fabricloader: Fabric Loader 0.14.22
fabric-api: Fabric API 0.87.0+1.20.1

```
java.lang.Error: Watchdog
        at java.base@17.0.8/java.lang.Object.wait(Native Method)
        at java.base@17.0.8/java.lang.Thread.join(Thread.java:1304)
        at java.base@17.0.8/java.lang.Thread.join(Thread.java:1372)
        at com.github.electroluxv2.utils.ScriptsManager.lambda$runOnSaveScript$1(ScriptsManager.java:95)
        at com.github.electroluxv2.utils.ScriptsManager$$Lambda$8946/0x00007ffa051e2340.run(Unknown Source)
        at net.minecraft.class_1255.method_20493(class_1255.java:78)
        at com.github.electroluxv2.utils.ScriptsManager.runOnSaveScript(ScriptsManager.java:93)
        at net.minecraft.server.MinecraftServer.handler$zba000$backup-scripts$save(MinecraftServer.java:2307)
        at net.minecraft.server.MinecraftServer.method_3723(MinecraftServer.java:553)
        at net.minecraft.server.MinecraftServer.method_39218(MinecraftServer.java:560)
        at net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:834)
        at net.minecraft.server.MinecraftServer.handler$zcg000$carpet$modifiedRunLoop(MinecraftServer.java:8942)
        at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:650)
        at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:265)
        at net.minecraft.server.MinecraftServer$$Lambda$6459/0x00007ffa04d270f0.run(Unknown Source)
        at java.base@17.0.8/java.lang.Thread.run(Thread.java:833)
```

**Comment by ElectroluxV2** [comment]
Hi,
Did this crash happen at stop or at autosave?

**Comment by wereii** [comment]
I am sure this happened during autosave (so on-save.sh). 
My backup script sends some of the paramateres into a (named) pipe and that's it, though what I am noticing now is that maybe if, for whatever reason, the backup daemon on the other side does not consume from the pipe then it will block the backup script indefinitely, not sure if that is a problem.

**Comment by ElectroluxV2** [comment]
Yes, blocking backup script indefinitely is a problem, probably thats is the reason behind crash.
But I may change implementation to use separate thread pool to run these scripts, so this would result in postponded exception that could be gracefully handled.

**Comment by ElectroluxV2** [comment]
Resolved in 1.2.0
